tests/scripts/all.sh

```diff
@@ -3824,6 +3824,88 @@ component_test_psa_crypto_config_reference_cipher_aead () {
     tests/compat.sh -V NO -p mbedTLS
 }
 
+common_block_cipher_dispatch() {
+    TEST_WITH_DRIVER="$1"
+
+    # Start from the full config
+    helper_libtestdriver1_adjust_config "full"
+
+    if [ "$TEST_WITH_DRIVER" -eq 1 ]; then
+        # Disable key types that are accelerated (there is no legacy equivalent
+        # symbol for ECB)
+        scripts/config.py unset MBEDTLS_AES_C
+        scripts/config.py unset MBEDTLS_ARIA_C
+        scripts/config.py unset MBEDTLS_CAMELLIA_C
+    fi
+
+    # Disable cipher's modes and AEADs that, when not accelerated, cause
+    # legacy key types to be re-enabled in "config_adjust_legacy_from_psa.h".
+    # Keep this also in the reference component in order to skip the same tests
+    # that were skipped in the accelerated one.
+    scripts/config.py -f "$CRYPTO_CONFIG_H" unset PSA_WANT_ALG_CTR
+    scripts/config.py -f "$CRYPTO_CONFIG_H" unset PSA_WANT_ALG_CFB
+    scripts/config.py -f "$CRYPTO_CONFIG_H" unset PSA_WANT_ALG_OFB
+    scripts/config.py -f "$CRYPTO_CONFIG_H" unset PSA_WANT_ALG_CBC_NO_PADDING
+    scripts/config.py -f "$CRYPTO_CONFIG_H" unset PSA_WANT_ALG_CBC_PKCS7
+    scripts/config.py -f "$CRYPTO_CONFIG_H" unset PSA_WANT_ALG_CMAC
+    scripts/config.py -f "$CRYPTO_CONFIG_H" unset PSA_WANT_ALG_CCM_STAR_NO_TAG
+    scripts/config.py -f "$CRYPTO_CONFIG_H" unset PSA_WANT_ALG_CCM
```

Comment:
Those two lines are very wrong: the whole point is that we want CCM and GCM to work here!

Reply:
Yes, it is true that we want CCM/GCM to work here, but this should happen through the legacy `cipher_block` interface; we're not checking the PSA path here. Moreover, as the comment above this block explains, enabling any of these symbols without acceleration will cause legacy key types (`MBEDTLS_(AES|ARIA|CAMELLLIA)_C`) to be re-enabled in `config_adjust_legacy_from_psa.h`.
So are you suggesting to accelerate also CCM and GCM here?

Reply:
> Edit: on a 2nd thought I think that I should update `config_adjust_legacy_from_psa.h` because even in case of builtin CCM/GCM we can rely on PSA dispatching of the block_cipher module. So please ignore the above comment for now while I will try to implement what you proposed.

Ah great, that's what I was going to say! The whole point of G5 is to be able to enable `PSA_WANT_ALG_[CG]CM` without `AES_C` (Aria, Camellia) as long as we have `ACCEL_KEY_TYPE_AES` (Aria, Camellia) + `ACCEL_ALG_ECB_NO_PADDING`. So we shouldn't stop until that's possible and proven by tests.